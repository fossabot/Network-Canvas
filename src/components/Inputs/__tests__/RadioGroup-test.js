/* eslint-env jest */

import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import FieldComponents from '../fieldComponents';

const setup = (props) => {
  const mockProps = {
    label: 'Radio group',
    input: {
      value: null,
      name: 'radio_group'
    },
    options: ['1', '2', '3'],
    meta: {
      invalid: true,
      error: 'an error message',
    },
    ...props,
  };

  const component = shallow((
    <FieldComponents.RadioGroup store={createStore(() => {})} {...mockProps} />
  ));

  return component;
};

describe('<RadioGroup />', () => {
  it('should render', () => {
    const subject = setup();

    expect(subject).toMatchSnapshot();
  });

  it('should render a toggle for each option', () => {
    const subject = setup();

    expect(subject.find('RadioInput').length).toBe(3);
  });
});
