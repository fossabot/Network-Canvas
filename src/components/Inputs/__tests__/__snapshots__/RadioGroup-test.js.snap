exports[`<RadioGroup /> should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="radio-group">
    <div>
        Radio group
    </div>
    <div>
        <div>
            <label
                className="radio-group__item"
                htmlFor="undefined_1">
                <input
                    checked={false}
                    id="undefined_1"
                    name={undefined}
                    onClick={[Function]}
                    type="radio"
                    value="1" />
                 
                1
            </label>
        </div>
        <div>
            <label
                className="radio-group__item"
                htmlFor="undefined_2">
                <input
                    checked={false}
                    id="undefined_2"
                    name={undefined}
                    onClick={[Function]}
                    type="radio"
                    value="2" />
                 
                2
            </label>
        </div>
        <div>
            <label
                className="radio-group__item"
                htmlFor="undefined_3">
                <input
                    checked={false}
                    id="undefined_3"
                    name={undefined}
                    onClick={[Function]}
                    type="radio"
                    value="3" />
                 
                3
            </label>
        </div>
    </div>
    <div>
        an error message
    </div>
</div>,
  "nodes": Array [
    <div
      className="radio-group">
      <div>
            Radio group
      </div>
      <div>
            <div>
                  <label
                        className="radio-group__item"
                        htmlFor="undefined_1">
                        <input
                              checked={false}
                              id="undefined_1"
                              name={undefined}
                              onClick={[Function]}
                              type="radio"
                              value="1" />
                         
                        1
                  </label>
            </div>
            <div>
                  <label
                        className="radio-group__item"
                        htmlFor="undefined_2">
                        <input
                              checked={false}
                              id="undefined_2"
                              name={undefined}
                              onClick={[Function]}
                              type="radio"
                              value="2" />
                         
                        2
                  </label>
            </div>
            <div>
                  <label
                        className="radio-group__item"
                        htmlFor="undefined_3">
                        <input
                              checked={false}
                              id="undefined_3"
                              name={undefined}
                              onClick={[Function]}
                              type="radio"
                              value="3" />
                         
                        3
                  </label>
            </div>
      </div>
      <div>
            an error message
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <RadioGroup
        input={
                Object {
                        "value": null,
                      }
        }
        label="Radio group"
        meta={
                Object {
                        "error": "an error message",
                        "invalid": true,
                      }
        }
        options={
                Array [
                        "1",
                        "2",
                        "3",
                      ]
        }
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "input": Object {
            "value": null,
          },
          "label": "Radio group",
          "meta": Object {
            "error": "an error message",
            "invalid": true,
          },
          "options": Array [
            "1",
            "2",
            "3",
          ],
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="radio-group">
          <div>
                    Radio group
          </div>
          <div>
                    <div>
                              <label
                                        className="radio-group__item"
                                        htmlFor="undefined_1">
                                        <input
                                                  checked={false}
                                                  id="undefined_1"
                                                  name={undefined}
                                                  onClick={[Function]}
                                                  type="radio"
                                                  value="1" />
                                         
                                        1
                              </label>
                    </div>
                    <div>
                              <label
                                        className="radio-group__item"
                                        htmlFor="undefined_2">
                                        <input
                                                  checked={false}
                                                  id="undefined_2"
                                                  name={undefined}
                                                  onClick={[Function]}
                                                  type="radio"
                                                  value="2" />
                                         
                                        2
                              </label>
                    </div>
                    <div>
                              <label
                                        className="radio-group__item"
                                        htmlFor="undefined_3">
                                        <input
                                                  checked={false}
                                                  id="undefined_3"
                                                  name={undefined}
                                                  onClick={[Function]}
                                                  type="radio"
                                                  value="3" />
                                         
                                        3
                              </label>
                    </div>
          </div>
          <div>
                    an error message
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="radio-group">
          <div>
                    Radio group
          </div>
          <div>
                    <div>
                              <label
                                        className="radio-group__item"
                                        htmlFor="undefined_1">
                                        <input
                                                  checked={false}
                                                  id="undefined_1"
                                                  name={undefined}
                                                  onClick={[Function]}
                                                  type="radio"
                                                  value="1" />
                                         
                                        1
                              </label>
                    </div>
                    <div>
                              <label
                                        className="radio-group__item"
                                        htmlFor="undefined_2">
                                        <input
                                                  checked={false}
                                                  id="undefined_2"
                                                  name={undefined}
                                                  onClick={[Function]}
                                                  type="radio"
                                                  value="2" />
                                         
                                        2
                              </label>
                    </div>
                    <div>
                              <label
                                        className="radio-group__item"
                                        htmlFor="undefined_3">
                                        <input
                                                  checked={false}
                                                  id="undefined_3"
                                                  name={undefined}
                                                  onClick={[Function]}
                                                  type="radio"
                                                  value="3" />
                                         
                                        3
                              </label>
                    </div>
          </div>
          <div>
                    an error message
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RadioGroup
    input={
        Object {
            "value": null,
          }
    }
    label="Radio group"
    meta={
        Object {
            "error": "an error message",
            "invalid": true,
          }
    }
    options={
        Array [
            "1",
            "2",
            "3",
          ]
    }
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    } />,
}
`;
